.github/workflows/build.yml

```diff
@@ -0,0 +1,198 @@
+    branches: [ master ]
+    tags: [ 'v*' ]
+  pull_request:
+    branches: [ master ]
+jobs:
+  build:
+    strategy:
+      matrix:
+        program:
+        - higan-ui
+        - byuu
+        - icarus
+        os:
+        - name: ubuntu
+          version: latest
+        - name: windows
+          version: latest
+        - name: macos
+          version: latest
+    runs-on: ${{ matrix.os.name }}-${{ matrix.os.version }}
+    steps:
+    - uses: actions/checkout@v2
+    - name: Install Dependencies
+      if: matrix.os.name == 'ubuntu'
+      run: |
+        sudo add-apt-repository -y "deb http://archive.ubuntu.com/ubuntu `lsb_release -sc` main universe restricted multiverse"
+        sudo apt-get update -y -qq
+        sudo apt-get install libsdl2-dev libgtksourceview2.0-dev libgtk2.0-dev libao-dev libopenal-dev
+    - name: Make
+      run: make -j4 -C ${{ matrix.program }} build=performance local=false
+    - name: Upload
+      uses: actions/upload-artifact@v2
+      with:
+        name: ${{ matrix.program }}-${{ matrix.os.name }}
+        path: ${{ matrix.program }}/out/${{ matrix.program }}*
+
+  release:
+    if: github.event.action != 'pull_request'
+    runs-on: ubuntu-latest
+    needs:
+    - build
+    steps:
+    - uses: actions/checkout@v2
+      with:
+        path: 'src'
+    - name: Download Artifacts
+      uses: actions/download-artifact@v2
+      with:
+        path: 'bin'
+    - name: Package Artifacts
+      run: |
+        set -e
+        case ${GITHUB_REF} in
+          refs/tags/*) suffix="-${GITHUB_REF#refs/tags/}" ;;
+          refs/heads/master) suffix="-nightly" ;;
+          *) suffix="" ;;
+        esac
+
+        srcdir="${GITHUB_WORKSPACE}/src"
+        bindir="${GITHUB_WORKSPACE}/bin"
+
+        for os in ubuntu windows macos
+        do
+          mkdir "${os}"
+          cd "${os}"
+
+          # Package higan.
+          outdir=higan${suffix}
+          mkdir ${outdir}
+          mkdir ${outdir}/Systems
+          cp -ar ${bindir}/higan-ui-${os}/* ${outdir}
+          cp -ar ${bindir}/icarus-${os}/* ${outdir}
+          cp -a ${srcdir}/higan/System ${outdir}/Templates
+          cp -a ${srcdir}/icarus/Database ${outdir}
+          cp -a ${srcdir}/icarus/Firmware ${outdir}
+          cp -a ${srcdir}/GPLv3.txt ${outdir}
+          cp -a ${srcdir}/extras/higan/* ${outdir}
+          cp -a ${srcdir}/extras/both/* ${outdir}
+          zip -r ../higan-${os}.zip ${outdir}
+
+          # Package byuu.
+          outdir=byuu${suffix}
+          mkdir ${outdir}
+          mkdir ${outdir}/Firmware
+          cp -ar ${bindir}/byuu-${os}/* ${outdir}
+          cp -a ${srcdir}/GPLv3.txt ${outdir}
+          cp -a ${srcdir}/extras/byuu/* ${outdir}
+          cp -a ${srcdir}/extras/both/* ${outdir}
+          zip -r ../byuu-${os}.zip ${outdir}
+
+          cd -
+        done
+    - name: Create Release
+      id: release
+      env:
+        GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+      run: |
+        set -e
```

Comment:
Another potential place for `-u` if appropriate.

Reply:
Done.